Development/phase2

johnhargrove opened the pull request:
Large PR with:

- Organization-owned Directory, Service, Service Policy, Public Key, SDK Key management
- Directory-owned Service, Service Policy, Public Key, SDK Key management
- Complete implementation of `.feature` file integration tests via SpecFlow, all tests passing
- **NOTE 1:** the integration test project is likely to cause AppVeyor build failures right now, as we didn't have AppVeyor working on the dev branch
- **NOTE 2:** the integration test project **will not work** unless `Secrets/OrgPrivateKey.txt` and `Secrets/OrgId.txt` files are present.

aenglander commented on src/iovation.LaunchKey.Sdk.Tests.Integration/Features/directory_client/services/directory-client-service-create.feature:
```diff
@@ -0,0 +1,26 @@
+
+  Scenario: Client can send unique Service name and successfully create a Service
+    When I create a Directory Service with the following:
+	  | key          | value                             |
```
White space issue. Using tabs instead of spaces

johnhargrove replied:
This and other white space issues should be resolved in my latest commits.

aenglander commented on src/iovation.LaunchKey.Sdk.Tests.Integration/Features/directory_client/services/directory-client-service-get-one.feature:
```diff
@@ -0,0 +1,25 @@
+
+  Scenario: Client can retrieve a single Service
+    Given I created a Directory Service with the following:
+	  | key          | value                             |
```
Spacing again

aenglander commented on src/iovation.LaunchKey.Sdk.Tests.Integration/Features/directory_client/services/directory-client-service-update.feature:
```diff
@@ -0,0 +1,31 @@
+  Background:
+    Given I created a Directory
+    And I created a Directory Service with the following:
+	  | key          | value                             |
```
Spacing again

aenglander commented on src/iovation.LaunchKey.Sdk.Tests.Integration/Features/directory_client/services/directory-client-service-update.feature:
```diff
@@ -0,0 +1,31 @@
+
+  Scenario: Client can update all Directory attributes other than ID and name
+    When I update the Directory Service with the following:
+	  | key          | value                                |
```
Spacing again

aenglander commented on src/iovation.LaunchKey.Sdk.Tests/FactoryFactoryBuilderTests.cs:
```diff
@@ -84,10 +84,20 @@ public void SetCrypto_ShouldReturnSelf()
 
 		[TestMethod]
-		public void SetCurrentPublicKeyTtl_ShouldReturnSelf()
+		public void SetCurrentPublicKeyTttl_ShouldReturnSelf()
```
Typo

johnhargrove replied:
This isn't a typo -- it's a test we must maintain. Will explain in your other Q on this.

aenglander commented on src/iovation.LaunchKey.Sdk.Tests/FactoryFactoryBuilderTests.cs:
```diff
@@ -84,10 +84,20 @@ public void SetCrypto_ShouldReturnSelf()
 		{
 			var ffb = new FactoryFactoryBuilder();
+#pragma warning disable CS0612
 			var result = ffb.SetCurrentPublicKeyTttl(10);
```
What us this all about. Why does PublicKeyTttl exist?

aenglander replied:
Okay

johnhargrove commented:
To answer your question regarding SpecFlow libraries, they should only be referenced in the testing projects. The tests that SpecFlow generates rely on these assemblies to (a) integrate with MSTest, and (b) provide the attributes, types and execution pipeline for feature files.

It should **not** affect our distribution.

johnhargrove commented:
My last couple of commits involved:

1) Create a new build configuration specific to AppVeyor. This is a release mode build with one change: The integration tests project is not built. 
2) Updated the AppVeyor configuration to use the new `Release_AppVeyor` configuration.
3) Updated the AppVeyor configuration to only run tests in the unit test project.

This should make sure the AppVeyor build is able to execute without the presence of the secrets.

johnhargrove commented:
Are we good to merge this? I see Brad and Devin have not had a chance to review yet. Unsure if the practice here is "everyone approves" or "at least 1 approves".

aenglander commented:
We definitely need Brad to review. He is the project lead.

Adam Englander
Software Architect 
O	+1 702.701.0106 
M	+1 702.267.8257 
 <https://twitter.com/iovation> <https://www.linkedin.com/in/adamenglander>
iovation.com <https://www.iovation.com/>
 <https://www.iovation.com/>



> On Sep 27, 2018, at 12:54 PM, John Hargrove <notifications@github.com> wrote:
> 
> Are we good to merge this? I see Brad and Devin have not had a chance to review yet. Unsure if the practice here is "everyone approves" or "at least 1 approves".
> 
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub <https://github.com/iovation/launchkey-dotnet/pull/5#issuecomment-425221870>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AAUDyPNQqLElD07cCoBgFxqvLnP0PnXUks5ufS0CgaJpZM4W3zTZ>.
>

bporter125 commented on src/iovation.LaunchKey.Sdk/Transport/Domain/DirectoryKeysPatchRequest.cs:
```diff
@@ -0,0 +1,28 @@
+		[JsonProperty("active")]
+		public bool Active { get; }
+
+		public DirectoryKeysPatchRequest(Guid serviceId, string keyId, DateTime? expires, bool active)
```
Looks like this param should be directoryId

johnhargrove commented:
Fixed the typos Brad identified.

Removed some old test code from the ExampleCli project that was causing the build to fail when this branch merged with @aenglander's latest changes on master. This was a code path covered by integration testing and wasn't needed, so I just removed it. AppVeyor build is now OK.

I hand this back to @aenglander @bporter125 @degan to approve and merge -- I think we're good here.

johnhargrove commented:
Actually, please standby, I'd like to bump the version number before you merge.
